ansible_builder/containerfile.py

```diff
@@ -211,10 +218,47 @@ def _create_folder_copy_files(self):
                 copy_file(str(script_path), scripts_dir)
 
         # later steps depend on base image containing these scripts
-        context_dir = pathlib.Path(self.build_outputs_dir).stem
+        context_dir = Path(self.build_outputs_dir).stem
         self.steps.append(f'COPY {context_dir}/scripts/ /output/scripts/')
 
+    def _handle_additional_build_files(self):
+        """
+        Deal with any files the user wants added to the image build context.
+
+        The 'src' value is either an absolute path, or a path relative to the
+        EE definition file. For example, 'src' can be a relative path like
+        "data_files/configs/*.cfg", but cannot be "/home/user/files/*.cfg",
+        the latter not being relative to the EE.
+        """
+        for entry in self.definition.additional_build_files:
+            src = Path(entry['src'])
+            dst = entry['dest']
+
+            # 'src' is either an absolute path or a path glob relative to the EE file
+            ee_file = Path(self.definition.filename)
+            if src.is_absolute():
```

Comment:
I'm thinking we need one more option here: a simple recursive dir-copy. Since we're just placing things into the generated context (not the actual containers yet), the simplest option (which the current impl doesn't support) is a recursive dir copy. So maybe if `src` is a dir, just blind copy the whole thing to `dest` with `shutil.copytree` or `os.walk` (if we want to preserve the ability to hook things like validating symlink targets and skipping apparently unmodified targets).

Reply:
The solution I chose was to iterate the `src` directory and call `copy_file()` for each child file (and recursively for any subdirectories). This might be slower, but attempts to preserve the build cache. I didn't want different caching behaviors based on the value of `src`.